damnit/ctxsupport/ctxrunner.py

```diff
@@ -345,6 +345,15 @@ def execute(self, run_data, run_number, proposal, input_vars) -> 'Results':
                             name, type(data)
                         )
                         data = None
+                    elif not np.issubdtype(arr.dtype, np.number):
+                        try:
+                            h5py.h5t.py_create(arr.dtype, logical=True)
+                        except TypeError:
+                            log.error("Variable %s returned %s which cannot "
+                                      "be converted to an HDF5 datatype object",
+                                      name, type(data)
```

Comment:
I think the error message here should mention the dtype of the array, because the type will usually be something like ndarray, or maybe list, so it could be confusing to say that can't be converted to HDF5.

Reply:
Good point. I changed the message to include both types, making clear the dtype may be the result of some form of conversion. For non-collection types the actual type may be more useful, while for collection types listing both may make it more likely to make the connection.